### PR TITLE
Filter locations in API by `location_on_map` field

### PR DESCRIPTION
**Pull request by jonbulz**
### Short description
Change location_not_on_map attribute of POIs to location_on_map and use it in api endpoints.

### Resolved issues

Fixes: #1038

### CODE REVIEW

**Review by timoludwig — COMMENTED** [review]
Awesome, congratz to your first PR :tada: :rocket: :heart_eyes: 

In general, this looks very good and does exactly what we want. :+1:
I have a few ideas for improvement:
1. We could filter the locations in the database query instead of the python loop - see my comment below
2. Changes to the database structure require migration files to ensure that all existing databases can adapt to the new structure. In your local repository clone, there should appear a new file in the `integreat_cms/cms/migrations/` directory which you need to add to your commit as well. If no such file was generated automatically, please execute the dev tool `./dev-tools/migrate.sh`. See the [Django documentation](https://docs.djangoproject.com/en/3.2/topics/migrations/) for more information.

**Comment by jonbulz** [comment]
Actually I think I messed around a little too much with the migration files. I wasn't sure what they were doing at first, so I did a bit of unnecessary back and forth editing. I tried to get things as clean as possible, but I always ended up creating at least 2 migration files. Is that right?

File: integreat_cms/cms/migrations/0016_rename_location_not_on_map_poi_location_on_map.py
Comment (timoludwig) [inline integreat_cms/cms/migrations/0016_rename_location_not_on_map_poi_location_on_map.py:18]:
```suggestion
        migrations.AlterField(
            model_name='poi',
            name='location_on_map',
            field=models.BooleanField(default=True, help_text='Tick if you want to show this location on map', verbose_name='Show this location on map'),
        ),
    ]
```

> Actually I think I messed around a little too much with the migration files. I wasn't sure what they were doing at first, so I did a bit of unnecessary back and forth editing. I tried to get things as clean as possible, but I always ended up creating at least 2 migration files. Is that right?


You can combine the two migration files by moving the operation of the second file into the first one and then deleting the second file.

File: integreat_cms/cms/migrations/0016_rename_location_not_on_map_poi_location_on_map.py
Comment (timoludwig) [inline integreat_cms/cms/migrations/0016_rename_location_not_on_map_poi_location_on_map.py:3]:
```suggestion
from django.db import migrations, models
```

File: integreat_cms/cms/migrations/0016_rename_location_not_on_map_poi_location_on_map.py
Comment (timoludwig) [inline integreat_cms/cms/migrations/0016_rename_location_not_on_map_poi_location_on_map.py:7]:
```suggestion
    """
    Migration file to rename the "location on map" field of the POI model
    """

```

**Review by timoludwig — COMMENTED** [review]
Thanks for your changes! :muscle: 

Apart from my comment below, we also have to fix the [CircleCI build for this PR](https://app.circleci.com/pipelines/github/digitalfabrik/integreat-cms/4189/workflows/85801963-325e-474d-b520-9e293bc5fa07):

At the moment, there are two problems:
1. The black code style is not applied - you can do this with the dev tool `./dev-tools/black.sh` - in the future, I recommend using [our pre-commit hooks](https://digitalfabrik.github.io/integreat-cms/git-flow.html#pre-commit-hooks) for this (which can be installed via `pipenv run pre-commit install`), but if you face any issues with them, you might as well execute the dev tool manually before you commit. 
2. The pylint code linter finds a few problems - a missing import and two missing docstrings. The docstring should be fixed with my suggestion, and for the import you need to add the line:
    ```
    from distutils.util import strtobool
    ```
    to the top of the locations api file as stated in my previous comment.

File: integreat_cms/cms/migrations/0016_rename_location_not_on_map_poi_location_on_map.py
Comment (timoludwig) [inline integreat_cms/cms/migrations/0016_rename_location_not_on_map_poi_location_on_map.py:12]:
```suggestion
        ("cms", "0020_alter_last_updated_field"),
```
In the meantime, there have been a few other migrations - and to ensure a linear migration history, we have to update the prior migration here. Also, could you increase the counter of this file and rename it to `0021_rename_location_not_on_map.py`? Thanks!

File: integreat_cms/api/v3/locations.py
Comment (timoludwig) [inline integreat_cms/api/v3/locations.py]:
```suggestion
from distutils.util import strtobool
from django.conf import settings
from django.http import JsonResponse
```
This should fix the pylint warning

**Review by timoludwig — COMMENTED** [review]
Thanks! 
I have one small suggestion left, and then this PR is almost ready to be merged :tada:
After that, please resolve the conflicts with the base branch by executing
```
git checkout develop
git pull
git checkout feature/location-on-map
git rebase develop
./dev-tools/resolve_translation_conflicts.sh
git rebase --continue
git push --force-with-lease
```
Then, after that it would be cool if you could clean up the git history a bit and combined your commits into one:
```
git rebase --interactive $(git merge-base --fork-point origin/develop $(git rev-parse --abbrev-ref HEAD))
# in the editor, replace "pick" by "fixup" in every line except the first one
git rebase --continue
git push --force-with-lease
```

File: integreat_cms/cms/migrations/0021_rename_location_not_on_map_poi_location_on_map.py
Comment (timoludwig) [inline integreat_cms/cms/migrations/0021_rename_location_not_on_map_poi_location_on_map.py:1]: I think this filename is a bit cumbersome, maybe `0021_rename_location_not_on_map.py` would be sufficient?

**Review by timoludwig — COMMENTED** [review]
Awesome, thanks a lot! :muscle: 

Oh sorry, and one thing I forgot: 
Could you add exactly one line at the bottom of the "UNRELEASED" of our [CHANGELOG](https://github.com/digitalfabrik/integreat-cms/blob/develop/CHANGELOG.md) with a note of this change? Thanks!
(If you don't want to repeat the git-clean-process, you can just use `git commit --no-edit --amend` together with `git push --force-with-lease`)

File: integreat_cms/cms/migrations/0021_rename_location_not_on_map_poi_location_on_map.py
Comment (jonbulz) [reply integreat_cms/cms/migrations/0021_rename_location_not_on_map_poi_location_on_map.py:1]: sure, hang on :) i had some issues with the translation file and pre-commit hooks anyway, but I think I fixed in now

**Comment by jonbulz** [comment]
> 

okay, I think I'm done :)

File: CHANGELOG.md
Comment (timoludwig) [inline CHANGELOG.md:14]:
```suggestion
* [ [#1038](https://github.com/digitalfabrik/integreat-cms/issues/1038) ] Rename location not on map attribute
```
If there's an issue associated with our PR, it's better to directly link the issue id (only fallback to the PR id if no issue exists)

**Review by timoludwig — COMMENTED** [review]
Thanks! Just a tiny little detail, but I think that's it :slightly_smiling_face:

File: CHANGELOG.md
Comment (jonbulz) [reply CHANGELOG.md]: sure, I mistook the pr with the issue id

**Review by timoludwig — COMMENTED** [review]
Damn, I just wanted to approve this PR, when I noticed that I used the old field recentlly when changing the POI coordinates...

Could you change this line

https://github.com/digitalfabrik/integreat-cms/blob/62d6128aa860ffdbf1b133de24d629f92930aaac/integreat_cms/cms/forms/pois/poi_form.py#L99
to:
```
 if cleaned_data.get("location_on_map"): 
```
?
Sorry about all the small incremental change requests :unamused:

**Comment by jonbulz** [comment]
> 

You're good, I didn't notice either. :D

**Review by timoludwig — APPROVED** [review]
Amazing, I think that's it :tada: :clinking_glasses: 
Now press that merge button before I find something else :laughing:

**Comment by jonbulz** [comment]
:champagne:

**Pull request merged**
